src/nwb_conversion_tools/tools/roiextractors/roiextractors.py

```diff
@@ -503,23 +503,32 @@ def image_mask_iterator():
             )
 
             ps = image_segmentation.create_plane_segmentation(**input_kwargs)
-        # Fluorescence Traces:
+
+        # Fluorescence Traces - This should be a function on its own.
         if "Flourescence" not in ophys.data_interfaces:
             fluorescence = Fluorescence()
             ophys.add(fluorescence)
         else:
             fluorescence = ophys.data_interfaces["Fluorescence"]
-        roi_response_dict = segext_obj.get_traces_dict()
+
         roi_table_region = ps.create_roi_table_region(
             description=f"region for Imaging plane{plane_no_loop}",
             region=list(range(segext_obj.get_num_rois())),
         )
+
+        roi_response_dict = segext_obj.get_traces_dict()
+
         rate = np.nan if segext_obj.get_sampling_frequency() is None else segext_obj.get_sampling_frequency()
-        for i, j in roi_response_dict.items():
-            data = getattr(segext_obj, f"_roi_response_{i}")
-            if data is not None:
-                data = np.asarray(data)
-                trace_name = "RoiResponseSeries" if i == "raw" else i.capitalize()
+
+        # Filter empty data
+        roi_response_dict = {key: value for key, value in roi_response_dict.items() if value is not None}
```

Comment:
A bit fancier/more elegant approach to filtering iterables in Python:

```python
roi_response_dict = dict(filter(lambda elem: elem[1] is not None, roi_response_dict .items()))
```

`elem` is an `item` in the `dict`, and `elem[1]` is the value (`elem[0]` is the key)

Reply:
Not sure it looks more elegant to me for this case. I think filter makes sense to me if the filter is more complex.

Reply:
Just wanted you to know about it - another plus is that `filter` is a lazy generator, so it doesn't actually 'apply' the filter until you call an executable on it. This doesn't make a difference here but it can in the future, and can be nice if the thing you are filtering is large.